sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -69,6 +69,14 @@ public HttpHeaders(Iterable<HttpHeader> headers) {
         }
     }
 
+    HttpHeaders(HttpHeaders headers) {
+        this.headers = new HashMap<>((int) (headers.headers.size() / 0.75f));
```

Comment:
is .75f a good number?

Reply:
Took this from the default load factor of a HashMap:

```
As a general rule, the default load factor (.75) offers a good tradeoff between time and space costs. Higher values decrease the space overhead but increase the lookup cost (reflected in most of the operations of the HashMap class, including get and put). The expected number of entries in the map and its load factor should be taken into account when setting its initial capacity, so as to minimize the number of rehash operations. If the initial capacity is greater than the maximum number of entries divided by the load factor, no rehash operations will ever occur.
```